stdlib/public/runtime/Errors.cpp

```diff
@@ -144,18 +144,16 @@ static bool getSymbolNameAddr(llvm::StringRef libraryName,
 void swift::dumpStackTraceEntry(unsigned index, void *framePC,
                                 bool shortOutput) {
 #if SWIFT_STDLIB_SUPPORTS_BACKTRACE_REPORTING && SWIFT_STDLIB_HAS_DLADDR
-  SymbolInfo syminfo;
-
-  // 0 is failure for lookupSymbol
-  if (0 == lookupSymbol(framePC, &syminfo)) {
+  auto syminfo = SymbolInfo::lookup(framePC);
```

Comment:
I'm not sure `auto` is great here - you don't have any indication that `SymbolInfo::lookup` returns a `SymbolInfo`.  Similar throughout.

Reply:
It doesn't, it returns an `llvm::Optional<SymbolInfo>`. This use of `auto` is not unheard of in the repo and is akin to Swift's `let` (well, `var`) which also doesn't explicitly state the type of the value.

Do you feel strongly about changing it?